Dockerfile

```diff
@@ -99,7 +84,20 @@ RUN if test -e modules/reana-commons; then \
       else \
         pip install --no-cache-dir "modules/reana-commons[snakemake_reports]" --upgrade; \
       fi \
-    fi
+    fi && \
+    apt-get remove -y \
+        cmake \
+        g++ \
+        gcc \
+        graphviz-dev \
+        libssl-dev \
+        make \
+        pkg-config \
+        python3.12-dev \
+        uuid-dev && \
+    apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
```

Comment:
Why is the removal of these packages being moved from the previous `RUN` step to this one? If the removal of packages is not done in the same `RUN` command as when they are installed, then those packages will increase the size of the docker image anyway, as all layers need to be downloaded (see for example https://stackoverflow.com/questions/61990329/benefits-of-repeated-apt-cache-cleans)

If we need some package here when installing `reana-commons`, then we should do something like
```
RUN apt-get update -y && \
  apt-get install -y --no-install-recommends <package to install> && \
  ... && \
  apt-get remove -y <package installed before> && \
  apt-get clean && ...
```